### PR TITLE
Add topics endpoint

### PR DESCRIPTION
**Pull request by jasonbarrows**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+describe('GET /api/topics', () => {
+  it('200: responds with all topics including slug and description properties', () => {
+    return request(app)
+      .get('/api/topics')
+      .expect(200)
+      .then(({ body }) => {
+        const { topics } = body;
+
+        expect(topics).toBeInstanceOf(Array);
+        expect(topics).toHaveLength(3);
```
Comment (Jodiech) [inline __tests__/app.test.js:19]: Nice to see you are checking the length here before the forEach 👍

File: controllers/topics.controller.js [hunk 2]
```diff
@@ -0,0 +1,7 @@
+const { selectAllTopics } = require("../models/topics.model");
+
+exports.getAllTopics = (req, res) => {
+  selectAllTopics().then((topics) => {
+    res.status(200).send({ topics });
+  });
```
Comment (Jodiech) [inline controllers/topics.controller.js]: Great to see you are returning the data on an appropriate key. Consider adding a catch block here too! 😄

File: __tests__/app.test.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+        topics.forEach((topic) => {
+          expect(topic).toMatchObject({
+            slug: expect.any(String),
+            description: expect.any(String),
+          });
+        });
+      });
+  });
+});
```
Comment (Jodiech) [inline __tests__/app.test.js:34]: Make sure to add a test for a status 404, for example: api/bananas

File: models/topics.model.js [hunk 4]
```diff
@@ -0,0 +1,9 @@
+const db = require('../db/connection');
+
+exports.selectAllTopics = () => {
+  return db.query(
+    `SELECT slug, description FROM topics;`
```
Comment (Jodiech) [inline models/topics.model.js]: You could change this to `SELECT * FROM topics`, it's the same and looks cleaner! 👍

**Review by Jodiech — COMMENTED** [review]
Hey 👋 nice work on this ticket! You are using the appropriate jest matchers in your test 👍 and you are checking the length assertion too! I've left a comment for you to add a test for a 404. Please make amendments and submit another PR when you're done 😄

File: models/topics.model.js (hunk 4 above)
Comment (jasonbarrows) [reply models/topics.model.js:5]: Thanks Jodie.

Would I add the test for a 404 into the same describe block for `describe('GET /api/topics')`?

It doesn't feel related.

File: __tests__/app.test.js (hunk 3 above)
Comment (jasonbarrows) [reply __tests__/app.test.js:34]: Thanks Jodie.

Would I add the test for a 404 status into the same describe block for `describe('GET /api/topics')`?

It doesn't feel related.

**Review by pippaamy — APPROVED** [review]
great changes, go ahead and merge!

**Pull request merged**